mlflow/async/client.py

```diff
@@ -0,0 +1,374 @@
+        if artifact_path:
+            rel_path = f"{artifact_path}/{rel_path}"
+
+        resp = await self._request(
+            "POST",
+            "/artifacts/credentials-for-write",
+            json={"run_id": run_id, "path": [str(rel_path)]},
+        )
+        credential_info = CredentialInfo(**resp["credential_infos"][0])
+        await self._upload_artifact(credential_info, local_path)
+
+    async def log_text(self, run_id: str, text: str, artifact_file: str) -> None:
+        with tempfile.TemporaryDirectory() as tmpdir:
+            splits = artifact_file.rsplit("/", 1)
+            if len(splits) == 2:
+                artifact_path, artifact_name = splits
+            else:
+                artifact_path = None
+                artifact_name = splits[0]
+
+            tmp_file = Path(tmpdir).joinpath(artifact_name)
+            tmp_file.write_text(text)
+
+            await self.log_artifact(run_id, tmp_file, artifact_path)
+
+    async def search_experiments(
+        self,
+        view_type: str = "ACTIVE_ONLY",
+        max_results: int = 100,
+        filter_string: Optional[str] = None,
+    ) -> List[Run]:
+        resp = await self._request(
+            "POST",
+            "/experiments/search",
+            params={
+                "view_type": view_type,
+                "max_results": max_results,
+                "filter": filter_string or "",
+            },
+        )
+        return [Experiment(**run) for run in resp.get("experiments", [])]
+
+
+@contextlib.contextmanager
+def timer():
+    start = time.time()
+    try:
+        yield
+    finally:
+        print(f"Took {time.time() - start:.2f} seconds")
+
+
+def load_host_and_token():
+    host = None
+    token = None
+    for l in (Path.home() / ".databrickscfg").read_text().splitlines():
+        if l.startswith("host"):
+            host = l.split("=")[1].strip()
+        if l.startswith("token"):
+            token = l.split("=")[1].strip()
+    assert host is not None
+    assert token is not None
+    return host, token
+
+
+async def main():
+    host, token = load_host_and_token()
+    name = "/Users/harutaka.kawamura@databricks.com/async-mlflow"
+    async with AsyncMlflowClient(host, token) as client:
+        experiment = await client.get_or_create_experiment(name)
+        async with client.start_run(experiment.experiment_id) as run:
+            with timer():
+                # Parallelize logging operations
+                await asyncio.gather(
+                    client.log_params(run.info.run_id, {"param1": "value1"}),
+                    client.log_metrics(run.info.run_id, {"accuracy": 0.9}),
+                    client.set_tags(run.info.run_id, {"tag1": "value1"}),
+                    client.log_artifact(run.info.run_id, __file__),
+                    client.log_artifacts(run.info.run_id, Path(__file__).parent, "dir"),
+                    client.log_text(run.info.run_id, "Hello, world!", "test.txt"),
+                )
```

Comment:
This took about 1 seconds on my machine. Doing the same with the existing sync APIs took 5 seconds. I'm not sure if this is a fair comparison.